File: Backend/core_functionalities/training_management_commands/src/models/strava_activity.py
from ..extensions import db

class StravaActivity(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    athlete_id = db.Column(db.Integer, index=True)
    name = db.Column(db.String(255))
    distance = db.Column(db.Float)  # in meters
    moving_time = db.Column(db.Integer)  # in seconds
    elapsed_time = db.Column(db.Integer)  # in seconds
    total_elevation_gain = db.Column(db.Float)  # in meters
    type = db.Column(db.String(50))

    def to_dict(self):
        return {
            "id": self.id,
            "athlete_id": self.athlete_id,
            "name": self.name,
            "distance": self.distance,
            "moving_time": self.moving_time,
            "elapsed_time": self.elapsed_time,
            "total_elevation_gain": self.total_elevation_gain,
            "type": self.type
        }

File: mobile/app/src/main/java/com/example/sportapp/data/model/StravaActivity.kt
package com.example.sportapp.data.model

data class StravaActivity(
    val id: Long,
    val athlete_id: Long,
    val name: String,
    val distance: Float,
    val moving_time: Int,
    val elapsed_time: Int,
    val total_elevation_gain: Float,
    val type: String
)

data class AccessTokenResponse(
    val token_type: String,
    val expires_at: Long,
    val expires_in: Int,
    val refresh_token: String,
    val access_token: String,
    val athlete: Athlete
)

data class Athlete(
    val id: Long, // Cambiado de Int a Long
    val username: String?,
    val resource_state: Int,
    val firstname: String,
    val lastname: String,
    val bio: String?,
    val city: String,
    val state: String,
    val country: String,
    val sex: String,
    val premium: Boolean,
    val summit: Boolean,
    val created_at: String,
    val updated_at: String,
    val badge_type_id: Int,
    val weight: Float?,
    val profile_medium: String,
    val profile: String,
    val friend: String?,
    val follower: String?
)
